cost/terminate_policy/server_terminate.pt

```diff
@@ -0,0 +1,188 @@
+      if ( !instance.locked ){
+        var instance_timestamp = instance.tags.split("=")[1];
+        var instance_terminate_date = new Date(instance_timestamp);
+        console.log(typeof(instance_terminate_date))
+        var instance_terminate_date_iso = instance_terminate_date
+        console.log(instance_terminate_date+"-"+iso_date)
+        var should_terminate = false
+        if (  instance_terminate_date <= iso_date ) {
+          console.log("true")
+          var should_terminate = true
+        }
+        
+        results.push({
+          resource_uid: instance.resource_uid,
+          name: instance.name,
+          href: instance.href
+          cloud_href: instance.cloud_href,
+          instance_type_href: instance.instance_type_href,
+          locked: instance.locked,
+          state: instance.state,
+          tags: instance.tags,
+          run_time: iso_date.toISOString(),
+          terminate: should_terminate
+        })
+      }
+    }
+EOF
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+escalation "report_and_terminate_instances" do
+  request_approval  do
+    label "Escalation approval"
+    description "Approve Terminate action"
+    parameter "approval_reason" do
+      type "string"
+      label "Reason for approval"
+      description "Explain why you are approving the action"
+    end
+  end
+
+  run "terminate_instances", data
+  email $param_email
+end
+
+resolution "report_resolution" do
+  email $param_email
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_instances" do
+  validate_each $ds_instances_with_timestamp do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Terminable Servers"
+    detail_template <<-EOS
+# Unattached IP Addresses
+| Name | Terminate_Tag | Current Time |
+| ---- | ------------- | ------------ |
+{{ range data -}}
+| {{ .name }} | {{.tags}} | {{.run_time}} |
+{{ end -}}
+EOS
+
+    escalate $report_and_terminate_instances
+    check logic_not(val(item,"terminate"))
+    resolve $report_resolution
+  end
+end
+
+###############################################################################
+# Cloud Workflow
+###############################################################################
+
+define terminate_instances($data) do
+  foreach $item in $data do
+    if !$item["locked"]
+      @instance = rs_cm.get(href: $item["href"])
+      @current_server = @instance.parent()
+      task_label("terminating instance - "+$item["name"])
+      @instance.terminate()
+    end
+  end
+end
+
+define sys_log($subject, $detail) do
+  if $$debug
+    rs_cm.audit_entries.create(
+      notify: "None",
+      audit_entry: {
+        auditee_href: @@account,
```

Comment:
the subject should include the policy name so it's easy to find in the audit entries.